solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -451,6 +451,67 @@ contract LiquidityProtection is TokenHandler, Utils, Owned, ReentrancyGuard, Tim
         return addProtectedLiquidity(_owner, poolToken, _baseToken, poolTokenAmount / 2, _amount);
     }
 
+    /**
+     * @dev returns the single-side staking limits of a given pool
+     *
+     * @param _poolAnchor   anchor of the pool
+     * @return maximum amount of base tokens that can be single-side staked in the pool
+     * @return maximum amount of network tokens that can be single-side staked in the pool
+     */
+    function poolAvailableSpace(IConverterAnchor _poolAnchor) external view returns (uint256, uint256) {
+        return (baseTokenAvailableSpace(_poolAnchor), networkTokenTokenAvailableSpace(_poolAnchor));
+    }
+
+    /**
+     * @dev returns the base-token staking limits of a given pool
+     *
+     * @param _poolAnchor   anchor of the pool
+     * @return maximum amount of base tokens that can be single-side staked in the pool
+     */
+    function baseTokenAvailableSpace(IConverterAnchor _poolAnchor) internal view returns (uint256) {
+        // get the pool converter
+        ILiquidityPoolConverter converter = ILiquidityPoolConverter(payable(ownedBy(_poolAnchor)));
+
+        // get the base token
+        IERC20Token baseToken = converter.connectorTokens(0);
+        if (baseToken == networkToken) {
+            baseToken = converter.connectorTokens(1);
+        }
+
+        // get the reserve balances
+        (uint256 reserveBalanceBase, uint256 reserveBalanceNetwork) =
+            converterReserveBalances(converter, baseToken, networkToken);
+
+        // get the network token minting limit
+        uint256 mintingLimit = settings.networkTokenMintingLimits(_poolAnchor);
+        if (mintingLimit == 0) {
+            mintingLimit = settings.defaultNetworkTokenMintingLimit();
+        }
+
+        // get the amount of network tokens minted for the pool
+        uint256 networkTokensMinted = settings.networkTokensMinted(_poolAnchor);
+
+        // return the maximum amount of base token liquidity that can be single-sided staked in the pool
+        return mintingLimit.sub(networkTokensMinted).mul(reserveBalanceBase).div(reserveBalanceNetwork);
```

Comment:
limit might be lower than the minted amount which will result in an exception.

Reply:
Of course, I think it is actually the current state on the ETHBNT pool.
But what do you wanna do about it, return 0 if `mintingLimit < networkTokensMinted`?

Reply:
yup

Reply:
Done.